src/main/scala/com/kinja/amqp/UnconfirmedMessageRepeater.scala

```diff
@@ -98,11 +110,15 @@ class UnconfirmedMessageRepeater(
 		transactional: TransactionalMessageStore,
 		republishTimeout: FiniteDuration
 	)(implicit ec: ExecutionContext): Unit = {
-		msgs.map { msg =>
+		msgs.foreach { msg =>
 			val result = Try(Await.result(producer.publish(msg.routingKey, Json.parse(msg.message)), republishTimeout))
 			result match {
 				case Success(_) => deleteMessageAndMatchingConfirm(msg, confs, transactional)
-				case Failure(ex) => logger.warn(s"""Couldn't resend message: $msg, ${ex.getMessage}""")
+				case Failure(ex: TimeoutException) =>
+					deleteMessageAndMatchingConfirm(msg, confs, transactional)
```

Comment:
If the publish times out, can we be sure that the new message will be saved to the msg store? What if there's a db connection problem?